### PR TITLE
code-review branch

### PR DESCRIPTION
**Pull request by Obaloluwa123**

### CODE REVIEW

File: app/src/main/java/com/example/fooding/activities/SignupActivity.java [hunk 1]
```diff
@@ -67,22 +68,22 @@ public void done(ParseException e) {
                     if (e == null) {
                         Log.i(TAG, "onClick signUp button");
                         Toast.makeText(getApplicationContext(),"Signed up sucessfully" , Toast.LENGTH_SHORT);
-
+                        goLoginActivity();
```
Comment (samkaks) [inline app/src/main/java/com/example/fooding/activities/SignupActivity.java:71]: same rename to openLoginActivity()

File: app/src/main/java/com/example/fooding/activities/SignupActivity.java [hunk 2]
```diff
@@ -67,22 +68,22 @@ public void done(ParseException e) {
...
                     }
 
                     else {
                         Log.i(TAG, "onClick Error");
                         Toast.makeText(getApplicationContext(),"Error Signing Up" , Toast.LENGTH_SHORT);
                         Log.e(TAG, e.toString());
                     }
-                    goLoginActivity();
+//                    goLoginActivity();
```
Comment (samkaks) [inline app/src/main/java/com/example/fooding/activities/SignupActivity.java:79]: don't commit comments unless they have a TODO

File: app/src/main/java/com/example/fooding/fragments/HomeFragment.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                try {
+                    JSONArray results = jsonObject.getJSONArray("results");
+                    Log.i(TAG, "Results: " + results.toString());
+                    Foods.addAll(Food.fromJsonArray(results));
+                    Foodadapter.notifyDataSetChanged();
+                    Log.i(TAG, "food: " + Foods.size());
+                } catch (JSONException e) {
+                    Log.e(TAG, "Hit json exception", e);
+                }
+            }
```
Comment (samkaks) [inline app/src/main/java/com/example/fooding/fragments/HomeFragment.java:107]: You probably want to refactor this out into the network call and add a callback here when the results are updated. Lmk if you need more help on doing this.

File: app/src/main/java/com/example/fooding/fragments/ProfileFragment.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+
+
+    public ProfileFragment() {
+        // Required empty public constructor
+    }
+
+
+
+
```
Comment (samkaks) [inline app/src/main/java/com/example/fooding/fragments/ProfileFragment.java:29]: unecessary white space

File: app/src/main/java/com/example/fooding/fragments/SearchFragment.java [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+//
+//                    @Override
+//                    public void onFailure(int statusCode, Headers headers, String response, Throwable throwable) {
+//                        Log.e(TAG,
+//                                "Request failed with code " + statusCode + ". Response message: " + response);
+//                    }
+//
+//
+//
+//
```
Comment (samkaks) [inline app/src/main/java/com/example/fooding/fragments/SearchFragment.java]: as i've mentioned before if you commit comments here, please leave a TODO

File: app/src/main/java/com/example/fooding/activities/LoginActivity.java [hunk 6]
```diff
@@ -29,10 +30,10 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_login);
         currentUser = ParseUser.getCurrentUser();
-
-//        if(currentUser != null){
-//            goMainActivity();
-//        }
+//
+        if(currentUser != null){
+            goMainActivity();
```
[the remark this answers is not in the document]
Comment (Obaloluwa123) [reply app/src/main/java/com/example/fooding/activities/LoginActivity.java:35]: I have renamed it

File: app/src/main/java/com/example/fooding/activities/SignupActivity.java (hunk 1 above)
Comment (Obaloluwa123) [reply app/src/main/java/com/example/fooding/activities/SignupActivity.java:71]: I have renamed it

File: app/src/main/java/com/example/fooding/activities/SignupActivity.java (hunk 2 above)
Comment (Obaloluwa123) [reply app/src/main/java/com/example/fooding/activities/SignupActivity.java:79]: I forgot to delete the comment, I won't commit that next time.

File: app/src/main/java/com/example/fooding/fragments/SearchFragment.java (hunk 5 above)
Comment (Obaloluwa123) [reply app/src/main/java/com/example/fooding/fragments/SearchFragment.java]: Noted

File: app/src/main/java/com/example/fooding/models/Food.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+public class Food {
+    String title;
+    String image;
+
+    public Food(JSONObject jsonObject) throws JSONException {
+        title = jsonObject.getString("title");
+        image = jsonObject.getString("image");
+
+
+
```
[the remark this answers is not in the document]
Comment (Obaloluwa123) [reply app/src/main/java/com/example/fooding/models/Food.java:18]: I have cleared the spaces

File: app/src/main/java/com/example/fooding/fragments/HomeFragment.java [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.example.fooding.adapters.FoodAdapter;
+import okhttp3.Headers;
+
+public class HomeFragment extends Fragment {
+    public static final String complex_search_url = "https://api.spoonacular.com/recipes/complexSearch?apiKey=f1bb97f5a6b141f1b5f8e17a2eba1296";
```
[the remark this answers is not in the document]
Comment (Obaloluwa123) [reply app/src/main/java/com/example/fooding/fragments/HomeFragment.java:41]: I have done that

File: app/src/main/java/com/example/fooding/activities/LoginActivity.java (hunk 6 above)
Comment (samkaks) [inline app/src/main/java/com/example/fooding/activities/LoginActivity.java]: renaming still left open

File: app/src/main/java/com/example/fooding/activities/SignupActivity.java (hunk 1 above)
Comment (samkaks) [reply app/src/main/java/com/example/fooding/activities/SignupActivity.java]: i don't see the update

File: app/src/main/java/com/example/fooding/fragments/HomeFragment.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (samkaks) [reply app/src/main/java/com/example/fooding/fragments/HomeFragment.java]: don't see the update still

**Review by samkaks — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/fooding/activities/LoginActivity.java (hunk 6 above)
Comment (Obaloluwa123) [reply app/src/main/java/com/example/fooding/activities/LoginActivity.java]: I updated it from my laptop, then pushed to GitHub

**Pull request merged**